### ISSUE TITLE
flutter run --machine outputs "event sent after app closed"

### ISSUE THREAD

**Issue by DanTup** [flutter/flutter#61361]
When running flutter `run --machine`, initial progress messages (if `pub get` runs) are output with an error:

```
danny@Dannys-MBP flutter_hello_world % /Users/danny/Dev/Google/flutter/bin/flutter --suppress-analytics run --machine --target lib/broken.dart -d flutter-tester --track-widget-creation --dart-define=flutter.inspector.structuredErrors=true --start-paused --web-server-debug-protocol ws --web-allow-expose-url 

event sent after app closed: {id: 0, progressId: null, message: Running "flutter pub get" in flutter_hello_world...}
event sent after app closed: {id: 0, progressId: null, finished: true}
[{"event":"daemon.connected","params":{"version":"0.5.3","pid":20044}}]
```

Because this isn't valid JSON, in VS Code we assume this is a message for the user and show it in the debug console (rather than showing a progress indicator).

![Screenshot 2020-07-14 at 10 50 25](https://user-images.githubusercontent.com/1078012/87411626-e7680200-c5bf-11ea-9950-80e7a583978f.png)

(@jonahwilliams sorry! 😬)

**Comment by guidezpl** [comment]
Also seeing this in Android Studio

**Comment by yvngaayush** [comment]
^me too

**Comment by devoncarew** [comment]
Yes, we're seeing this in IntelliJ as well:

<img width="942" alt="Screen Shot 2020-09-08 at 8 20 26 PM" src="https://user-images.githubusercontent.com/1269969/92619502-7323a500-f276-11ea-874a-f74d449da88e.png">

I believe some refactoring of the logging meant that we can no longer send progress events from the flutter tool early in the app's lifecycle. I have a local change which largely addresses this (but am not certain where the best place is to put a test). I'll follow up with a PR.

**Comment by devoncarew** [comment]
fixed by https://github.com/flutter/flutter/pull/65508

**Comment by Ayan4044** [comment]
I got this error too. Need to kill the dart.exe and then close the editor and reopen it

**Comment by zionnite** [comment]
internet connection make me received that error

**Comment by xJamieCarrollx** [comment]
I found that I was getting this error message while using flutter on Mac.

It seemed to be linked to the location the flutter sdk was located.

I originally put it in the applications folder and got these errors.

The solution for me was to move it to ~/Developer/Resources 2 directories I created in the home directory.

Then I manually modified the ~/.zshrc file to include this

export PATH=$PATH:~/Developer/Resources/flutter/bin

After all that I ran: " flutter run " inside one of my projects, and it worked.
